src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -7,7 +7,7 @@
 
 
 class DatasetFirebaseRepository:
-    WRITE_BATCH_SIZE = 500
+    WRITE_BATCH_SIZE = 100
```

Comment:
any reason batch size is reduced ?

Reply:
That is to avoid hitting the maximum size limit (10Mb) per write operation. We can have a method to dynamically control it depending on the unit_data size later.